crates/db-allocator/src/lib.rs

```diff
@@ -48,35 +48,87 @@ impl Constraint {
     }
 
     /// Set the min constraint.
-    pub fn min<T>(mut self, min: T) -> Self
+    pub fn min<T>(&mut self, min: T)
     where
         u64: From<T>,
     {
-        self.min = u64::from(min);
-        self
+        let min = u64::from(min);
+        if min > self.max {
```

Comment:
We should be very careful to use `panic()`, and avoid it if possible.

Reply:
fixed